cmd/operator-sdk/new/cmd.go

```diff
@@ -436,3 +462,26 @@ func initGit() error {
 	log.Info("Run git init done")
 	return nil
 }
+
+func validateProject() error {
+	log.Info("Validating project")
+	switch projutil.DepManagerType(depManager) {
+	case projutil.DepManagerGoMod:
+		// Run "go build ./..." to make sure all packages can be built
+		// currectly. From "go help build":
+		//
+		//	When compiling multiple packages or a single non-main package,
+		//	build compiles the packages but discards the resulting object,
+		//	serving only as a check that the packages can be built.
+		opts := projutil.GoCmdOptions{
+			PackagePath: "./...",
+			Dir:         filepath.Join(projutil.MustGetwd(), projectName),
+		}
+		if err := projutil.GoBuild(opts); err != nil {
+			return err
+		}
+	}
+
+	log.Info("Project validation successful.")
```

Comment:
Nit: I would just move this before }, so on line 483, if that makes sense. But just a nit, rather than anything else. :)